### DISCUSSION TITLE
Import repo from existing repo? Different than a fork?

### DISCUSSION THREAD

**Discussion by stuartv666** [community/community#27407] [New to GitHub]
I just created my first repo in my github account. 

I did it via a button on my github dashboard that said something like "Create repository" and let me choose to create it by importing an existing repo. I created my new repo by inputting a URL to another user's repo on github. This button does not appear to show on my github dashboard anymore, now that I actually have a repo. But, it seems there is a link on a dropdown for "Import repository", which takes me to the same dialog for importing an existing project.

Is the end result different than if I had started by going to the other user's repo on github and clicking the button there to create a fork for myself?

I want to be able to submit PRs back to the original project and I am unclear on whether the process I used to create my own repo on GitHub will facilitate that. Do I need to delete the repo I just created and go create a new one by clicking the Fork button on the original project's page?

Everywhere I search for forking versus cloning or importing is providing answers on the basis that the person doing a clone is cloning from a repo to their local computer. I believe what I did was clone from one github repo to my own github repo. Next, I will clone from my github repo to a local instance. But, is whatever link from the original repo to my github repo there, like it would be if I had used the Fork button?

Thanks.

**Comment by xRodzMC** [comment]
When you fork a repository, you create a copy of the original repository (upstream repository) but the repository remains on your GitHub account. Whereas, when you clone a repository, the repository is copied on to your local machine with the help of Git.
![image](https://user-images.githubusercontent.com/100104970/181412538-f1d26fb1-5677-4b09-94fb-82b8168984a9.png)
For more information you can read more [here](https://www.toolsqa.com/git/difference-between-git-clone-and-git-fork/)
![image](https://user-images.githubusercontent.com/100104970/181412186-54eeff94-4c7b-4202-8317-2068ad3a7a74.png)

**Reply by stuartv666** [reply]
Right. That is exactly the information that I said was not answering my question.

I did an Import on Github, of another Github project. I didn't do anything that pulled any files onto my local computer. I did not do a clone (other than speculating that the Import that I did was actually a server-side clone behind the scenes).

I'm trying to find out if doing an Import of one github repo into my own github account is the same as doing a fork of that repo into my github account.

**Reply by stuartv666** [reply]
Meanwhile, I think I answered my own question. The Import was taking FOREVER.

I did a Fork on a different repo (that was a fork of the project I was Importing). The fork operation completed pretty much instantly.

I conclude that doing an Import was making a physical copy of the original, but not preserving any relationship info to the original to facilitate submitting a PR back to the original. Whereas, doing a Fork from the original does preserve that relationship info.

So, I think I have answered my own question. All good now, unless somebody else chimes in and corrects me.

**Reply by jfullstackdev** [reply]
not very sure of your `create new repo` and import, is it a local one ? 

these are all the imports on GitHub 

https://docs.github.com/en/get-started/importing-your-projects-to-github

**Reply by stuartv666** [reply]
I did what is shown in the link you posted. The 3rd link on the page, that says "Importing a repository with GitHub Importer".

For the repository to import, I entered the URL to another GitHub repo. 

I was asking if doing that was the same as going to the other GitHub repo and creating a Fork into my own account.

I see now that it is not.

**Reply by xRodzMC** [reply]
The **Import Repository** is being used to import old project inaccessible to push. We will say its a clone project but with this feature you can determine that we cant make pull request to the original repository. **Fork repo** is using for example, making contribution such as pull request.

Its possible to DMCA takedown if you copying repo without permission as they own the code.

**Reply by jfullstackdev** [reply]
> but with this feature you can determine that we cant make pull request to the original repository

pull request is a feature of GitHub, even Git does not have that exact pull request, just a similar one

> Its possible to DMCA takedown if you copying repo without permission as they own the code.

like any other open source, it's up to the license of the project

and this 

https://docs.github.com/en/site-policy/content-removal-policies/dmca-takedown-policy

**Reply by xRodzMC** [reply]
> > but with this feature you can determine that we cant make pull request to the original repository
> 
> pull request is a feature of GitHub, even Git does not have that exact pull request, just a similar one
> 
> > Its possible to DMCA takedown if you copying repo without permission as they own the code.
> 
> like any other open source, it's up to the license of the project
> 
> and this
> 
> https://docs.github.com/en/site-policy/content-removal-policies/dmca-takedown-policy

However, we can say its depending on license. 

And i knew it. **Import Repository** is being used for importing other projects. For example, a gitlab project. This might helpful to import projects from other project services such as gitlab.
